docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![Supported Java versions](https://img.shields.io/badge/Java-11-blue.svg) 
+![Supported OS](https://img.shields.io/badge/Supported%20OS-Windows|MacOS|Linux-yellow.svg) 
 
+## Table of Contents
 
-## Product Scope
-### Target user profile
 
-{Describe the target user profile}
+- [1. Introduction](#1-introduction)
+    * [1.1. Purpose](#11-purpose)
+    * [1.2. Scope](#12-scope)
+- [2. Setting up](#2-setting-up)
+- [3. Design](#3-design)
+    * [3.1. Architecture](#31-architecture)
+    * [3.2. TextUi Component](#32-textui-component)
+    * [3.3. Logic Component](#33-logic-component)
+    * [3.4. Model Component](#34-model-component)
+    * [3.5. Storage Component](#35-storage-component)
+- [4. Implementation](#4-implementation)
+    * [4.1. Patient Details Feature](#41-patient-details-feature)
+        + [4.1.1. Add Patient Details](#411-add-patient-details)
+        + [4.1.2. Edit Patient Details](#412-edit-patient-details)
+        + [4.1.3. Delete Patient Details](#413-delete-patient-details)
+        + [4.1.4. List Patients](#414-list-patients)
+        + [4.1.5. Retrieve Patient Details](#415-retrieve-patient-details)
+    * [4.2. Patient Medical Records Feature](#42-patient-medical-records-feature)
+        
+    * [4.3. Appointment Scheduling Feature](#43-appointment-scheduling-feature)
+        + [4.3.1. Add Appointment](#431-add-appointment)
+        + [4.3.2. Edit Appointment](#432-edit-appointment)
+        + [4.3.3. Delete Appointment](#433-delete-appointment)
+        + [4.3.4. Mark Appointment as Done](#434-mark-appointment-as-done)
+        + [4.3.5. List Appointments](#435-list-appointments)
+        + [4.3.6. Find Appointments of Patient](#436-find-appointments-of-patient)
+    * [4.4 Storage](#44-storage)
+    * [4.5 Logging](#45-logging)
+- [5. Documentation](#5-documentation)
+- [6. Testing](#6-testing)
+- [7. Useful Links](#7-useful-links)
+- [Appendices](#appendices)
+    * [Appendix A: Product Scope](#qppendix-a-product-scope)
+    * [Appendix B: User Stories](#appendix-b-user-stories)
+    * [Appendix C: Non-functional Requirements](#appendix-c-non-functional-requirements)
+    * [Appendix D: Glossary](#appendix-d-glossary)
+    * [Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+## 1. Introduction
 
-## User Stories
+### 1.1. Purpose
+
+This document contains the architecture and software design specifications for the application HappyPills.
+
+### 1.2. Scope
+
+This guide is mainly for developers, designers and software engineers that are working on and using HappyPills.
+
+## 2. Setting Up
+
+1. Ensure that you have `Java 11` or later installed in your computer 
+2. Click [here](link to be added later?) to download the latest HappyPills JAR File
```

Comment:
I suggest linking to the jar file now that you released it! https://github.com/AY1920S2-CS2113T-T12-2/tp/releases/tag/v2.0